system_tests/ec2_handler.py

```diff
@@ -0,0 +1,211 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import copy
+import random
+from contextlib import contextmanager
+
+from boto.ec2 import EC2Connection
+from boto import config
+
+from cosmo_tester.framework.handlers import (
+    BaseHandler,
+    BaseCloudifyInputsConfigReader)
+
+
+class EC2CleanupContext(BaseHandler.CleanupContext):
+
+    def __init__(self, context_name, env):
+        super(EC2CleanupContext, self).__init__(context_name, env)
+        self.before_run = self.env.handler.ec2_infra_state()
+
+    def cleanup(self):
+        super(EC2CleanupContext, self).cleanup()
+        resources_to_teardown = self.get_resources_to_teardown()
+        if self.skip_cleanup:
+            self.logger.warn('[{0}] SKIPPING cleanup: of the resources: {1}'
+                             .format(self.context_name, resources_to_teardown))
+            return
+        self.logger.info('[{0}] Performing cleanup: will try removing these '
+                         'resources: {1}'
+                         .format(self.context_name, resources_to_teardown))
+
+        leftovers = self.env.handler.remove_ec2_resources(
+            resources_to_teardown)
+        self.logger.info('[{0}] Leftover resources after cleanup: {1}'
+                         .format(self.context_name, leftovers))
+
+    def get_resources_to_teardown(self):
+        current_state = self.env.handler.ec2_infra_state()
+        return self.env.handler.ec2_infra_state_delta(
+            before=self.before_run, after=current_state)
+
+
+class CloudifyEC2InputsConfigReader(BaseCloudifyInputsConfigReader):
+
+    def __init__(self, cloudify_config, manager_blueprint_path, **kwargs):
```

Comment:
If you are not using the constructor, it would be better simply not overriding it in the first place.